example/viam_example_app/lib/screens/motor.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_platform_widgets/flutter_platform_widgets.dart';
+import 'package:viam_sdk/viam_sdk.dart';
+
+class MotorScreen extends StatefulWidget {
+  final Motor motor;
+  final ResourceName resourceName;
+
+  const MotorScreen({Key? key, required this.motor, required this.resourceName}) : super(key: key);
+
+  @override
+  State<MotorScreen> createState() {
+    return _MotorScreenState();
+  }
+}
+
+class _MotorScreenState extends State<MotorScreen> {
+  double power = 0;
+  bool isOn = false;
+
+  void moveMotor(double powerPct) {
+    widget.motor.setPower(powerPct / 100);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return PlatformScaffold(
+      appBar: PlatformAppBar(
+        title: Text(widget.resourceName.name.toUpperCase()),
+      ),
+      iosContentPadding: true,
+      body: Center(
+        child: Column(
+          children: [
+            const SizedBox(height: 16),
+            PlatformText(
+              '${widget.resourceName.namespace}:${widget.resourceName.type}:${widget.resourceName.subtype}/${widget.resourceName.name}',
+              style: const TextStyle(fontWeight: FontWeight.w300),
+            ),
+            const SizedBox(height: 16),
+            Column(
+              children: [
+                Text('Power %: $power'),
+                Slider(
+                  value: power,
+                  onChanged: (value) => setState(() {
+                    power = value;
+                  }),
+                  max: 100,
+                  min: -100,
+                  divisions: 200,
+                  label: '$power',
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    ElevatedButton(
+                      onPressed: () => moveMotor(power),
+                      child: const Text('Run'),
+                    ),
+                    const SizedBox(width: 16),
+                    ElevatedButton(
+                      style: ButtonStyle(backgroundColor: MaterialStateProperty.all<Color>(Colors.red)),
+                      onPressed: () => widget.motor.stop(),
+                      child: const Text('Stop'),
+                    ),
```

Comment:
This is really heavily nested, it makes it kinda hard for me to read/parse. I don't really know dart at all so I'm happy to defer if this is typical, but I wonder if there's a way we could factor chunks out so that this is a bit clearer of a workflow?

Reply:
This is a good callout, I considered doing this multiple times but in the end decided not to for this screen. In a larger more complex screen, I agree, breaking things out into chunks would be a must. because of the simplicity of this screen and the lack of reused widgets (the visual components) I decided not factor things out into their own chunks.

Flutter is naturally very nested, so it may just need some adjusting when you're looking at a flutter view vs a dart file. Happp to take this offline and walk you (or anyone) whats going on in this file, if that would help it make more sense.

Reply:
No need, appreciate the offer! I think I can follow along, just a little more eye-movement than usual :)

Reply:
I was thinking about this feedback this morning and looked through the PR again and realized there was an extra nested column that was unneeded. I removed it, and there is one less layer of nesting now. I am glad you brought this up 😄 ✊